src/main/scala/org/ergoplatform/nodeView/state/UtxoState.scala

```diff
@@ -66,32 +66,34 @@ class UtxoState(override val persistentProver: PersistentBatchAVLProver[Digest32
   @SuppressWarnings(Array("TryGet"))
   private[state] def applyTransactions(transactions: Seq[ErgoTransaction],
                                        expectedDigest: ADDigest,
-                                       height: Height) = Try {
-
-    val createdOutputs = transactions.flatMap(_.outputs).map(o => (ByteArrayWrapper(o.id), o)).toMap
-    val totalCost = transactions.map { tx =>
-      tx.statelessValidity.get
-      val boxesToSpend = tx.inputs.map(_.boxId).map { id =>
-        createdOutputs.get(ByteArrayWrapper(id)).orElse(boxById(id)) match {
-          case Some(box) => box
-          case None => throw new Error(s"Box with id ${Algos.encode(id)} not found")
+                                       height: Height) = persistentProver.synchronized {
```

Comment:
UtxoState is not thread-safe, and thus should encapsulated in NodeViewHolder instance (which is an actor). Doing synchronization on writes then does not make sense.

Reply:
This synchronization is done together with `proofsForTransactions` at https://github.com/ergoplatform/ergo/pull/386/files#diff-804b1a8751b1f609f0501f81261be3deR73. We do not need to lock whole `NodeViewHolder` during proof generation, but proof generation during block application may lead to incorrect proofs, and thus miner will generate incorrect block locally.